### PR TITLE
first review push

### PR DESCRIPTION
**Pull request by Dineshg49**

### CODE REVIEW

File: .github/workflows/deployment-script.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
             node-version: '12'
             check-latest: true
     - run: |
-        git clone -b pipeline-new https://github.com/virtual-labs/ph3-lab-mgmt
+        git clone https://github.com/virtual-labs/ph3-lab-mgmt
```
Comment (raj-vlabs) [inline .github/workflows/deployment-script.yml]: Please use '--depth 1' here

File: experiment/posttest.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[{"question": "1. We apply the experiment in close end condition.", "answers": {"a": "True", "b": "False"}, "correctAnswer": "b"}, {"question": "2. Longitudinal stress is equal to half the hoop stress in same pressure.", "answers": {"a": "True", "b": "False"}, "correctAnswer": "b"}, {"question": "3. In an open end condition, longitudinal stress is equal to zero.", "answers": {"a": "True", "b": "False"}, "correctAnswer": "b"}]
```
Comment (raj-vlabs) [inline experiment/posttest.json]: I assume that we are waiting for the SME for the questions and the is the reason the questions are not following the new format

File: experiment/references.md [hunk 3]
```diff
@@ -1 +1,12 @@
-### Link your references in here
\ No newline at end of file
+
+
+  -  https://www.scribd.com/doc/136565/Mechanics-of-Materials-Torsion-Test
```
Comment (raj-vlabs) [inline experiment/references.md]: Please use the agreed upon format for references

File: experiment/simulation/css/main.css [hunk 4]
```diff
@@ -1 +1,24 @@
...
+	overflow-y: scroll;
+}
+
+.main {
+	width: 85%;
+	height: 85%;
+	background-color: #A4C652;
+}
+
+.action-button {
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: remove all unused classes

File: experiment/simulation/index.html [hunk 5]
```diff
@@ -1,13 +1,66 @@
...
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/virtual-labs/virtual-style@0.0.5-b/css/style.min.css"/>
+		<link rel="stylesheet" href="./css/main.css">
+	</head>
 
-	<!-- Add JS at the bottom of HTML file --> 
-        <script src="./js/main.js"></script>
-</body>
+		<div class="columns is-centered">
+			<div class="column has-text-centered mt-2">
+				<button id="play" class="action-button">Play</button>
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Please use classes from the virtual-style library

File: experiment/simulation/index.html [hunk 6]
```diff
@@ -1,13 +1,66 @@
...
+				<button id="pause" class="action-button">Pause</button>
+				<button id="restart" class="action-button">Restart</button>
+				<div class="slidecontainer">
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Where is slidecontainer defined?

File: experiment/simulation/js/main.js [hunk 7]
```diff
@@ -1 +1,280 @@
...
+
+		while(j<15)
+		{
+			let curr = [
+				[startX + width[j],mid - ht[j]],
+				[startX + width[j+1],mid-ht[j]],
+				[startX + width[j+1],mid+ht[j]],
+				[startX + width[j],mid+ht[j]]
+			]
+				if(flag==1)
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Please use ===

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
Many of the basic coding guidelines from DASS are not being followed. Please run the code against the DASS coding style guidelines before submitting for review

File: experiment/posttest.json (hunk 2 above)
Comment (Dineshg49) [reply experiment/posttest.json]: Only the Simulation Folder part of the code was done. This requires help from SME.

File: experiment/references.md (hunk 3 above)
Comment (Dineshg49) [reply experiment/references.md]: Only the Simulation Folder part of the code was done. This requires help from SME.

File: experiment/posttest.json [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+                "c": "Remains same",
+                "d": "None of these"
+            },
+            "correctAnswer": "a",
+            "difficulty": "intermediate"
+        },
+        {
+            "question": "The polar moment of inertia of a solid circular shaft of diameter (d) is",
+            "answers": {
+                "a": "π(d^2)/16",
```
Comment (raj-vlabs) [inline experiment/posttest.json]: Can we use superscript in place of ^ ?

File: experiment/pretest.json [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+                "a": "Compressive force",
+                "b": "Type of friction",
+                "c": "Twisting",
+                "d": "Object at rest"
+            },
+            "explanations": {
+                "c": "Torsion is defined as the twisting of a specimen caused by a certain amount of torque. Compressive forces are forces that are applied to an object that results in the specimen getting compacted."
+            },
+            "correctAnswer": "c",
+            "difficulty": "beginner"
```
Comment (raj-vlabs) [inline experiment/pretest.json:15]: Only beginner level questions available on pretest

File: experiment/procedure.md [hunk 10]
```diff
@@ -1 +1,25 @@
...
+6. Set the protector to zero for convenience and clamp it by means knurled screw.
+7. Carry out straining by rotating the hand wheel in either direction.
+8. Load the machine in suitable increments, taking note of the torque and the corresponding angle of twist.
+9. Plot a graph between torque and angle of twist and calculate the value of G by using the following relation
+
+   T/J = Gϴ/L <br>
+
+   Where <br>
+   T = Torque applied <br>
+   J = polar moment of inertia of the shaft = πd4/32 <br>
```
Comment (raj-vlabs) [inline experiment/procedure.md]: Does 4 represent a power of d? It should be in superscript then

File: experiment/simulation/index.html [hunk 11]
```diff
@@ -1,13 +1,64 @@
...
-<head>
-	<!-- Add CSS at the head of HTML file -->
-    <link rel="stylesheet" href="./css/main.css">
-</head>
-<body>
-		<!-- Your code goes here-->
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1">
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.1/css/bulma.min.css">
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Remove Bulma when using virtual-style

File: experiment/simulation/index.html [hunk 12]
```diff
@@ -1,13 +1,64 @@
...
+				</div>
+			  </div>
+			<div class="column has-text-centered">
+				<div id="chartContainer"></div>
+			</div>
+		</div>
+
+		<script src="https://unpkg.com/mathjs@9.2.0/lib/browser/math.js"></script>
+		<script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
+		<script src='https://cdn.plot.ly/plotly-latest.min.js'></script>
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Where is plotly used in this experiment?

File: experiment/theory.md [hunk 13]
```diff
@@ -1 +1,18 @@
...
+Imagine a horizontal radial line drawn on the end face. When the end is twisted, the line rotates through an angle θ.
+G is one of the elastic constants of the material. The equation is only true so long as the material remains elastic.
+<br>
+
+
+T/J = Gϴ/L <br>
+
+Where <br>
+T = Torque applied <br>
+J = polar moment of inertia of the shaft = πd4/32 <br>
```
Comment (raj-vlabs) [inline experiment/theory.md]: If 4 should be a power of d, please use it as superscript

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
Is this experiment responsive?

File: experiment/pretest.json (hunk 9 above)
Comment (Dineshg49) [reply experiment/pretest.json:15]: yes sir pranav said we could have 0 questions for some section if we want. and sme suggested that only the beginner level questions

File: experiment/simulation/index.html [hunk 14]
```diff
@@ -1,13 +1,63 @@
...
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="column has-text-centered">
+        <div id="chartContainer"></div>
+    </div>
+</div>
+
+<script src="https://canvasjs.com/assets/script/canvasjs.min.js"></script>
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Canvasjs needs to be replaced by plotly as discussed earlier

**Review by raj-vlabs — CHANGES_REQUESTED** [review]

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**